main.tf

```diff
@@ -279,42 +281,54 @@ resource "azurerm_linux_virtual_machine" "vm_linux" {
       condition     = alltrue([for value in var.admin_ssh_keys : value.username == var.admin_username || value.username == null])
       error_message = "`username` in var.admin_ssh_keys should be the same as `admin_username` or `null`."
     }
+    precondition {
+      condition     = var.bypass_platform_safety_checks_on_user_schedule_enabled == false ? true : local.patch_mode == "AutomaticByPlatform"
```

Comment:
If `var.bypass_platform_safety_checks_on_user_schedule_enabled` cannot be `null`, then this expression could be simplify to:

```hcl
condition     = !var.bypass_platform_safety_checks_on_user_schedule_enabled || local.patch_mode == "AutomaticByPlatform"
```